vpr/src/route/partition_tree.cpp

```diff
@@ -139,20 +149,65 @@ std::unique_ptr<PartitionTreeNode> PartitionTree::build_helper(const Netlist<>&
         for (auto net_id : nets) {
             t_bb bb = route_ctx.route_bb[net_id];
             if (bb.ymax < best_pos) {
-                left_nets.push_back(net_id);
+                left_nets.insert(net_id);
             } else if (bb.ymin > best_pos) {
-                right_nets.push_back(net_id);
+                right_nets.insert(net_id);
             } else {
-                my_nets.push_back(net_id);
+                my_nets.insert(net_id);
             }
         }
 
         out->left = build_helper(netlist, left_nets, x1, y1, x2, std::floor(best_pos));
         out->right = build_helper(netlist, right_nets, x1, std::floor(best_pos + 1), x2, y2);
     }
 
+    if(out->left)
+        out->left->parent = out.get();
+    if(out->right)
+        out->right->parent = out.get();
+
+    out->bb = {x1, x2, y1, y2, 0, 0};
     out->nets = my_nets;
     out->cutline_axis = best_axis;
     out->cutline_pos = best_pos;
+
+    /* Build net to ptree node lookup */
+    for(auto net_id: my_nets){
+        _net_to_ptree_node[net_id] = out.get();
+    }
     return out;
 }
+
+inline bool net_in_ptree_node(ParentNetId net_id, const PartitionTreeNode* node){
```

Comment:
This should either be in the .h file, or you should have a declaration in the .h file (if you don't already).  Make it private if it is just a class helper.  Also add Doxygen comment (returns true if nets bounding box is completely within the partition node's bounding box).